cylc/flow/ws_data_mgr.py

```diff
@@ -478,30 +488,34 @@ def generate_graph_elements(self, edges=None,
                 self.edge_points[s_point_cls].add(t_point_cls)
                 if target_id not in task_proxies:
                     task_proxies[target_id] = self.generate_ghost_task(t_node)
-                if target_id not in self.tasks[t_task_id].proxies:
-                    self.tasks[t_task_id].proxies.append(target_id)
+                if target_id not in tasks[t_task_id].proxies:
+                    tasks[t_task_id].proxies.append(target_id)
+
+                # Initiate edge element.
+                e_id = (
+                    f'{self.workflow_id}{ID_DELIM}{s_node}{ID_DELIM}{t_node}')
+                edges[e_id] = PbEdge(
+                    id=e_id,
+                    suicide=edge[3],
+                    cond=edge[4],
+                )
+                edges[e_id].source = source_id
                 edges[e_id].target = target_id
 
-        # If no edges were created (i.e. no source/target met criteria).
-        if not edges:
-            return
-        graph.edges.extend([e.id for e in edges.values()])
-        graph.leaves.extend(config.leaves)
-        graph.feet.extend(config.feet)
-        for key, info in config.suite_polling_tasks.items():
-            graph.workflow_polling_tasks.add(
-                local_proxy=key,
-                workflow=info[0],
-                remote_proxy=info[1],
-                req_state=info[2],
-                graph_string=info[3],
-            )
+                # Add edge id to node field for resolver reference
+                task_proxies[target_id].edges.append(e_id)
+                if s_valid:
+                    task_proxies[source_id].edges.append(e_id)
+
+        graph.edges.extend(edges.keys())
+
+        if new_points:
+            self.generate_ghost_families(family_proxies, new_points)
 
-        self.generate_ghost_families(family_proxies, new_points)
         # Replace the originals (atomic update, for access from other threads).
-        self.task_proxies = task_proxies
-        self.edges = edges
-        self.graph = graph
+        self.data[self.workflow_id]['task_proxies'] = task_proxies
+        self.data[self.workflow_id]['edges'] = edges
+        self.data[self.workflow_id]['graph'] = graph
```

Comment:
Worth using constants for some of these keys?

Reply:
You mean:
```
TASK_PROXIES='task_proxies'
.
.
.
        self.data[self.workflow_id][TASK_PROXIES] = task_proxies
```
at the class or module level?